contracts/token/ERC721/ERC721.sol

```diff
@@ -330,9 +330,32 @@ contract ERC721 is Context, ERC165, IERC721 {
         if (!to.isContract()) {
             return true;
         }
-
-        bytes4 retval = IERC721Receiver(to).onERC721Received(_msgSender(), from, tokenId, _data);
-        return (retval == _ERC721_RECEIVED);
+        bytes memory payload = abi.encodeWithSelector(
+            IERC721Receiver(to).onERC721Received.selector,
+            "onERC721Received(address,address,uint256,bytes)",
```

Comment:
This line should be removed, it was used as the first argument to `abi.encodeWithSignature`, but now we're using `abi.encodeWithSelector`.

```suggestion
```